src/index.ts

```diff
@@ -0,0 +1,43 @@
+import fetch from 'cross-fetch';
+import startsWith from 'lodash.startswith';
+
+type FetchClient = (
+  input: RequestInfo,
+  init?: RequestInit | undefined,
+) => Promise<Response>;
+
+async function checkResponseIsOk(response: Response): Promise<Response> {
+  if (response.ok) {
+    return response;
+  }
+  const error = await response.text();
+  return Promise.reject(error);
+}
+
+async function responseToCompletion(
+  response: Response,
+): Promise<string | object> {
+  const contentType = response.headers.get('Content-Type');
+  if (contentType && startsWith(contentType, 'application/json')) {
+    const text = await response.text();
+    try {
+      return JSON.parse(text);
+    } catch (e) {
+      return text;
```

Comment:
It's weird that you are handling the error here assuming the content-type is JSON already. I'd assume this function fails if the header says it's a JSON, but calling JSON.parse on the body fails. Shouldn't be the responsibility of the caller to handle errors like this one?